Homework3/Program.cs

```diff
@@ -228,7 +228,9 @@ static async Task Main()
             tests(text1, "second",5);
 
             tests(text2, "last", 5);
-            
+
+            // Comment for Github
+            //////
         }
```

Comment:
The purpose of this homework is to use interfaces for different concerns: 

- an interface for generating text (and 3 implementations)
- an interface for the search algo (and 2 implementations)

Can you restructure the homework to use this (so 2 new interfaces and 5 new concrete classes).

Reply:
For me is complicate to realize what the interfaces and the clasees are need to create.

Reply:
Ok, hai ca propun eu interfetele: 


```csharp
public interface ITextGenerator
{
    Task<string> FetchTextAsync(); 
    // sau daca vrei fara async, despre care o sa vorbim maine string FetchText();
}

public interface IStringSearcher
{
    int Search(string haystack, string needle);
}
```

Si apoi cateva clase gen

```csharp
class BoyerMooreSearch : IStringSearcher  { .... }
class LibraryStringSearch : IStringSearcher  { .... }

class InteretText : ITextGenerator { ... }
class SimpleText : ITextGenerator { ... }
```

Incearca sa muti logica in clasele astea si sa rescrii programul.